Code/Framework/AzCore/AzCore/DOM/DomBackendRegistryInterface.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomBackend.h>
+
+namespace AZ::DOM
+{
+    //! Central interface for registering and looking up backend types by file extension.
+    class BackendRegistryInterface
+    {
+    public:
+        AZ_RTTI(BackendRegistryInterface, "{B72A88AC-DF15-4F92-9489-ABCDEA3D94E6}")
+        using BackendFactory = AZStd::function<AZStd::unique_ptr<Backend>()>;
+
+        virtual ~BackendRegistryInterface() = default;
+
+        //! Registers a factory for a given backend.
+        //! \param name A unique name to identify this backend.
+        //! \param extensions A set of file extensions this backend supports.
+        //! Extensions should by the entire file suffix including any dots (.), for example:
+        //! ".json" or ".tar.gz"
+        template<class TBackend>
+        void RegisterBackend(AZStd::string name, AZStd::vector<AZStd::string> extensions);
+
+        //! Looks up a DOM backend based on its name.
+        virtual AZStd::unique_ptr<Backend> GetBackendByName(AZStd::string_view name) = 0;
+
+        //! Looks up a DOM backend based on a file extension.
+        virtual AZStd::unique_ptr<Backend> GetBackendForExtension(AZStd::string_view extension) = 0;
```

Comment:
This is conflating what the implementation of a backend can do with the implementations might ultimately want.
I find it strange that it prescribes some kind of mapping notion of file extension to backend.
What if there a backend for writing over the network or to the VFS.

I don't think there should be mapping of file extensions to backend and only a unique name.

Reply:
I've removed the registry for now, we can dig into the design a bit more if and when we find the need for a central way to look up DOM backends - for now their interface will still be useful, and we can have bespoke file-extension mappings on a per-system basis.